src/visualization/ColorPalette/ColorMap.cc

```diff
@@ -0,0 +1,284 @@
+#include <string>
+#include <vector>
+#include <SFML/Graphics/Color.hpp>
+#include <array>
+#include <cmath>
+#include "ColorMap.h"
+#include <iostream>
+
+using namespace std;
+using namespace sf;
+
+/**
+ * This Method will compute in terms of the argument sequences and the k, all kmers we are looking for and call
+ * the method which compute us in terms of this number a Colormap. Then the method will compute
+ * all different Kmers and push them in a List. We'll use this list to initialize differentKmers.
+ */
+ColorMap::ColorMap(vector<string>& Sequences, int k){
+    ///< This is a variable which save the number of Kmers per Sequence which we have
+    vector<int> SequenceSizes;
+    for (int i=0;i<Sequences.size();i++)
+        SequenceSizes[i] = (int)floor((Sequences[i].size())/k);
+
+    ///< This is a variable which save the number of Kmers
+    int numbOfKmer = 0;
+    for (int i=0;i<SequenceSizes.size();i++)
+        numbOfKmer += SequenceSizes[i];
+    updateColorList(numbOfKmer);
+    ///< This variable is the output List
+    vector<string> kmerDeclarations;
+
+    ///< a boolean for the "got this kmer a color?"-Check
+    bool matched = false;
+    ///< a index for the k-mers per sequence
+    int index;
+    
+    /**
+     * This part of the method will push all Kmers in the List, which are not in the list already.
+     */
+    for (int i=0;i<Sequences.size();i++) {
+        index = 0;
+        while (index<SequenceSizes[i]) {
+            ///< Comparing all k-mers to know if the Kmer is already in the list
+            for (int j=0;j<kmerDeclarations.size();j++) {
+                if (kmerDeclarations[j]==Sequences[i].substr(index*k, k)) {
```

Comment:
This is duplicated functionality. The set of nodes of the graph has to be determined from the sequences elsewhere.
This requires
 1) to identify duplicates of kmers and
 2) to remove all kmers which do not occur at least in two sequences
It is not trivial and should not be duplicated.